pkg/monitoring/vmstat/hyperv/provider_windows.go

```diff
@@ -0,0 +1,70 @@
+// +build windows
+
+package hyperv
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cloudradar-monitoring/cagent/perfcounters"
+	"github.com/cloudradar-monitoring/cagent/pkg/monitoring"
+	"github.com/cloudradar-monitoring/cagent/pkg/monitoring/vmstat"
+	"github.com/cloudradar-monitoring/cagent/pkg/wmi"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func init() {
+	prov := newImpl()
+
+	err := vmstat.RegisterVMProvider(prov)
+	if err != nil {
+		panic(err.Error())
```

Comment:
should we really panic?

Reply:
I think so as it happens at init stage and doing some logging is not proper here. This panic is more for develop and test purpose.
And to prevent duplicated entries in config (which may be cause of panic because such vm provider already exists) we should do config validation after reading file

Reply:
> This panic is more for develop and test purpose.
This sounds odd.

Still, causing panic inside a package is no good practice. The error should bubble up and `log.Fatal` or something similar should be called on a higher level then.